src/transformers/generation/utils.py

```diff
@@ -2083,7 +2083,7 @@ def generate(
         # - `model_kwargs` may be updated in place with a cache as defined by the parameters in `generation_config`.
         # - different models have a different cache name expected by the model (default = "past_key_values")
         # - `max_length`, prepared above, is used to determine the maximum cache length
-        max_cache_length = generation_config.max_length
+        max_cache_length = generation_config.max_length - 1
```

Comment:
We were creating caches that were larger than what was set by our max length flags: the last token is not present in the cache.